src/mmvc/base/TriggerMap.hx

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2012 Massive Interactive
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of 
+this software and associated documentation files (the "Software"), to deal in 
+the Software without restriction, including without limitation the rights to 
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies 
+of the Software, and to permit persons to whom the Software is furnished to do 
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all 
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+SOFTWARE.
+*/
+
+package mmvc.base;
+
+import haxe.ds.EnumValueMap;
+import haxe.ds.ObjectMap;
+import haxe.ds.StringMap;
+
+import minject.Injector;
+
+import mmvc.api.ICommand;
+import mmvc.api.ITriggerMap;
+
+class TriggerMap implements ITriggerMap
+{
+	var injector:Injector;
+	var classToCommand:StringMap<Class<ICommand>>;
+	var stringToCommand:StringMap<Class<ICommand>>;
+	var enumValueToCommand:EnumValueMap<EnumValue, Class<ICommand>>;
+	var objectToCommand:ObjectMap<Dynamic, Class<ICommand>>;
+
+	public function new(injector:Injector)
+	{
+		this.injector = injector;
+		classToCommand = new StringMap<Class<ICommand>>();
+		stringToCommand = new StringMap<Class<ICommand>>();
+		enumValueToCommand = new EnumValueMap<EnumValue, Class<ICommand>>();
+		objectToCommand = new ObjectMap<Dynamic, Class<ICommand>>();
+	}
+
+	public function map(trigger:Class<Dynamic>, command:Class<ICommand>)
+	{
+		var triggerClassName = Type.getClassName(trigger);
+		if(classToCommand.exists(triggerClassName))
+			throw "Command for class " + triggerClassName + " is already mapped.";
+
+		classToCommand.set(triggerClassName, command);
+	}
+
+	public function unmap(trigger:Class<Dynamic>)
+	{
+		classToCommand.remove(Type.getClassName(trigger));
+	}
+
+	public function dispatch(trigger:{})
+	{
+		var triggerClass = Type.getClass(trigger);
```

Comment:
I think we need to use `Type.typeof` to check whether trigger is a class instances here.